Userland/Libraries/LibGfx/ImageFormats/ILBMLoader.cpp

```diff
@@ -307,6 +337,16 @@ static ErrorOr<void> decode_bmhd_chunk(ILBMLoadingContext& context)
         return Error::from_string_literal("IFFImageDecoderPlugin: Not enough data for header chunk");
 
     context.bm_header = *bit_cast<BMHDHeader const*>(first_chunk.data.data());
+
+    if (context.bm_header.planes > 8)
```

Comment:
I couldn't find any documentation which places a hard limit (below 255) on the number of bitplanes, but we do calculations later on that fail if the number of bitplanes is > 8. I also couldn't find any examples of files that use more than 8 bitplanes.

[This document](https://web.archive.org/web/20140613000343/http://home.comcast.net/~erniew/lwsdk/docs/filefmts/ilbm.html) suggests that bitplanes may be ignored if they are not required, so we could possibly do that instead.

Reply:
There can be files with 24 bitplanes actually. They are called "deep" and contain 24 bitplanes with full RGB data.

There can be a mask bitplane as well but I am not sure it is supported for deep files.

Reply:
I have opened a [PR](https://github.com/SerenityOS/serenity/pull/21324) to add HAM support.

24bit support was my next task.

Reply:
I think this check is probably worth keeping for now. I've updated the error message to be a bit more meaningful. We can update/remove the check when support for deep ILBMs is added.